test/unit/pcv/PCVDepositAggregator.test.ts

```diff
@@ -0,0 +1,444 @@
+
+    beforeEach(async () => {
+      const tokenFactory = new MockERC20__factory(impersonatedSigners[userAddress]);
+      const pcvDepositAggregatorDeployer = new PCVDepositAggregator__factory(impersonatedSigners[userAddress]);
+      const mockPCVDepositDeployer = new MockPCVDepositV2__factory(impersonatedSigners[userAddress]);
+
+      token = await tokenFactory.deploy();
+      await token.deployTransaction.wait();
+
+      pcvDeposit = await mockPCVDepositDeployer.deploy(core.address, token.address, ethers.utils.parseEther('1000'), 0);
+      await pcvDeposit.deployTransaction.wait();
+
+      pcvDepositAggregator = await pcvDepositAggregatorDeployer.deploy(
+        core.address,
+        core.address,
+        token.address,
+        [pcvDeposit.address],
+        [90],
+        10
+      );
+      await pcvDepositAggregator.deployTransaction.wait();
+    });
+
+    it('returns expected values for deposits, weights, balances, and token', async () => {
+      await token.mint(pcvDeposit.address, ethers.utils.parseEther('1000'));
+      expect(await pcvDepositAggregator.getTotalBalance()).to.equal(ethers.utils.parseEther('1000'));
+      expect(await pcvDepositAggregator.paused()).to.be.false;
+      expect(await pcvDepositAggregator.bufferWeight()).to.be.equal(10);
+      expect(await pcvDepositAggregator.token()).to.equal(token.address);
+    });
+
+    it('successfully rebalances', async () => {});
+  });
+
+  describe('when it is deployed with multiple deposits', async () => {
+    let pcvDepositAggregator: PCVDepositAggregator;
+    let token: MockERC20;
+    let pcvDeposit1: PCVDeposit;
+    let pcvDeposit2: PCVDeposit;
+    let pcvDeposit3: PCVDeposit;
+
+    beforeEach(async () => {
+      const tokenFactory = new MockERC20__factory(impersonatedSigners[userAddress]);
+      const pcvDepositAggregatorDeployer = new PCVDepositAggregator__factory(impersonatedSigners[userAddress]);
+      const mockPCVDepositDeployer = new MockPCVDepositV2__factory(impersonatedSigners[userAddress]);
+
+      token = await tokenFactory.deploy();
+      await token.deployTransaction.wait();
+
+      pcvDeposit1 = await mockPCVDepositDeployer.deploy(core.address, token.address, 0, 0);
+      await pcvDeposit1.deployTransaction.wait();
+
+      pcvDeposit2 = await mockPCVDepositDeployer.deploy(core.address, token.address, 0, 0);
+      await pcvDeposit2.deployTransaction.wait();
+
+      pcvDeposit3 = await mockPCVDepositDeployer.deploy(core.address, token.address, 0, 0);
+      await pcvDeposit3.deployTransaction.wait();
+
+      pcvDepositAggregator = await pcvDepositAggregatorDeployer.deploy(
+        core.address,
+        core.address,
+        token.address,
+        [pcvDeposit1.address, pcvDeposit2.address, pcvDeposit3.address],
+        [20, 30, 40],
+        10
+      );
+      await pcvDepositAggregator.deployTransaction.wait();
+    });
+
+    it('returns expected values for deposits, weights, balances, and token', async () => {
```

Comment:
This is similar to the Init functionality normally used but also has some other logic around the depositing behavior